File: 0118-pascals-triangle/0118-pascals-triangle.java
// T- O(n2)
class Solution {
    public static List<Integer> generateRow(int n){
        int ans = 1;
        List<Integer> ls = new ArrayList<>();
        ls.add(ans);
        for (int i = 1; i<n; i++){
            ans = ans*(n-i);
            ans = ans/i;
            ls.add(ans);
        }
        return ls;
    }
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ret = new ArrayList<>();
        for (int i = 1; i<=numRows; i++){
            ret.add(generateRow(i));
        }
        return ret;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        initlst = [[1]]
        for i in range(1,numRows):
            l = []
            lst = initlst[i-1]
            for j in range(i+1):
                if j-1>=0 and j<len(lst):
                    l.append(lst[j-1]+lst[j])
                else:
                    l.append(1)
            initlst.append(l)
            print("pass ",i)
            print(initlst)
        return initlst
